wallet/wallet.go

```diff
@@ -3926,6 +3958,220 @@ func (w *Wallet) Database() walletdb.DB {
 	return w.db
 }
 
+// RemoveDescendants attempts to remove any transaction from the wallet's tx
+// store (that may be unconfirmed) that spends outputs created by the passed
+// transaction. This remove propagates recursively down the chain of descendent
+// transactions.
+func (w *Wallet) RemoveDescendants(tx *wire.MsgTx) error {
+	txRecord, err := wtxmgr.NewTxRecordFromMsgTx(tx, time.Now())
+	if err != nil {
+		return err
+	}
+
+	return walletdb.Update(w.db, func(tx walletdb.ReadWriteTx) error {
+		wtxmgrNs := tx.ReadWriteBucket(wtxmgrNamespaceKey)
+
+		return w.TxStore.RemoveUnminedTx(wtxmgrNs, txRecord)
+	})
+}
+
+// BirthdayBlock returns the birthday block of the wallet.
+//
+// NOTE: The wallet won't start until the backend is synced, thus the birthday
+// block won't be set and `ErrBirthdayBlockNotSet` will be returned.
+func (w *Wallet) BirthdayBlock() (*waddrmgr.BlockStamp, error) {
+	var birthdayBlock waddrmgr.BlockStamp
+
+	// Query the wallet's birthday block height from db.
+	err := walletdb.View(w.db, func(tx walletdb.ReadTx) error {
+		addrmgrNs := tx.ReadBucket(waddrmgrNamespaceKey)
+
+		bb, _, err := w.Manager.BirthdayBlock(addrmgrNs)
+		birthdayBlock = bb
+
+		return err
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return &birthdayBlock, nil
+}
+
+// AddScopeManager creates a new scoped key manager from the root manager.
+func (w *Wallet) AddScopeManager(scope waddrmgr.KeyScope,
+	addrSchema waddrmgr.ScopeAddrSchema) (
+	*waddrmgr.ScopedKeyManager, error) {
+
+	var scopedManager *waddrmgr.ScopedKeyManager
+
+	err := walletdb.Update(w.db, func(tx walletdb.ReadWriteTx) error {
+		addrmgrNs := tx.ReadWriteBucket(waddrmgrNamespaceKey)
+
+		manager, err := w.Manager.NewScopedKeyManager(
+			addrmgrNs, scope, addrSchema,
+		)
+		scopedManager = manager
+
+		return err
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return scopedManager, nil
+}
+
+// InitAccounts creates accounts with all possible key families, which ranges
+// from 1 to 255.
+func (w *Wallet) InitAccounts(scope *waddrmgr.ScopedKeyManager,
+	watchOnly bool) error {
+
+	return walletdb.Update(w.db, func(tx walletdb.ReadWriteTx) error {
+		addrmgrNs := tx.ReadWriteBucket(waddrmgrNamespaceKey)
+
+		// Generate all accounts that we could ever need. This includes
+		// all key families.
+		for keyFam := uint32(1); keyFam <= 255; keyFam++ {
```

Comment:
Perhaps just use the term account here since key family is an lnd term? May also want to consider accepting the upper limit of this loop as an arg.

Reply:
Updated.